data_model/src/transaction.rs

```diff
@@ -320,22 +349,9 @@ impl SignedTransactionV1 {
     }
 }
 
-impl TransactionValue {
-    /// Calculate transaction [`Hash`](`iroha_crypto::HashOf`).
-    pub fn hash(&self) -> iroha_crypto::HashOf<SignedTransaction> {
-        self.value.hash()
-    }
-
-    /// [`Transaction`] payload.
-    #[inline]
-    pub fn payload(&self) -> &TransactionPayload {
-        self.value.payload()
-    }
-
-    /// [`iroha_crypto::SignatureOf`]<[`TransactionPayload`]>.
-    #[inline]
-    pub fn signatures(&self) -> &SignaturesOf<TransactionPayload> {
-        self.value.signatures()
+impl AsRef<SignedTransaction> for TransactionValue {
```

Comment:
how will a user get `TransactionValue::error`? you have to make a getter function for that field

Reply:
Added `[getset]` macro

Reply:
Added `[getset]` macro